DNN Platform/Library/Entities/Portals/PortalController.cs

```diff
@@ -2815,19 +2815,24 @@ public static string DeletePortal(PortalInfo portal, string serverPath)
                         }
                     }
                     //delete upload directory
-                    Globals.DeleteFolderRecursive(serverPath + "Portals\\" + portal.PortalID);
                     if (!string.IsNullOrEmpty(portal.HomeDirectory))
                     {
-                        string homeDirectory = portal.HomeDirectoryMapPath;
-
-                        if (homeDirectory.EndsWith("\\"))
-                        {
-                            homeDirectory = homeDirectory.Substring(0, homeDirectory.Length - 1);
-                        }
-
-                        if (Directory.Exists(homeDirectory))
+                        var homeDirectory = portal.HomeDirectoryMapPath;
+                        // check whether home directory is not used by other portal
+                        // it happens when new portal creation failed, but home directory defined by user is alredy in use with other portal
+                        var homeDirectoryInUse = portals.OfType<PortalInfo>().Any(
+                            x => x.PortalID != portal.PortalID &&
+                            x.HomeDirectoryMapPath.Equals(homeDirectory, StringComparison.OrdinalIgnoreCase));

```

Comment:
At first I did not know what I was looking at. Maybe it will be a bit more readable as follows:

```c#
var homeDirectoryInUse = portals.OfType<PortalInfo>().Any(x => 
    x.PortalID != portal.PortalID &&
    x.HomeDirectoryMapPath.Equals(homeDirectory, StringComparison.OrdinalIgnoreCase));
```

Also please add a blank line before the `if`

Reply:
fixed